src/rdkafka_broker.c

```diff
@@ -4691,7 +4723,8 @@ static int rd_kafka_broker_thread_main(void *arg) {
 
                 if (RD_KAFKA_BROKER_IS_LOGICAL(rkb)) {
                         rd_atomic32_sub(&rkb->rkb_rk->rk_logical_broker_cnt, 1);
-                } else if (rkb->rkb_down_reported) {
+                } else if (rd_atomic32_set(&rkb->rkb_down_reported, 0) == 1) {
```

Comment:
Its confusing to change these values from multiple places. Can we try to change them from a single place? Either here or from state change?

I am bit confused here on how to reach here. We reach this condition only when `skip_broker_down` was set to true and broker handle is getting destroyed. Don't you think that we should be adding ALL_BROKERS_DOWN counter in this condition?